spores-pickling/src/test/scala/scala/spores/run/pickling/PicklingBinary.scala

```diff
@@ -12,72 +12,74 @@ import binary._
 
 import SporePickler._
 
-
 trait Emitter[T] {
   def emit(v: T)(implicit pickler: Pickler[T], unpickler: Unpickler[T]): Unit
   def done(): Unit
 }
 
 class TestEmitter extends Emitter[String] {
   val builder = new StringBuilder()
-  def emit(v: String)(implicit pickler: Pickler[String], unpickler: Unpickler[String]): Unit =
+
+  def emit(v: String)(implicit pickler: Pickler[String],
+                               unpickler: Unpickler[String]): Unit = 
     builder.append(v)
+
   def done(): Unit = {
     // do nothing
   }
+
 }
 
 @RunWith(classOf[JUnit4])
 class PicklingBinarySpec {
+
+  implicit val staticOnly = static.StaticOnly
+
   @Test
   def `pickle/unpickle to/from binary`(): Unit = {
+
     val v1 = 10
     val s = spore {
       val c1 = v1
       (x: Int) => s"arg: $x, c1: $c1"
     }
 
-    val pickler: Pickler[Spore[Int, String] { type Captured = Int }] with Unpickler[Spore[Int, String] { type Captured = Int }] =
-    SporePickler.genSporePickler[Int, String, Int]
-
-    val format = implicitly[PickleFormat]
-    val builder = format.createBuilder
-
-    builder.hintTag(implicitly[FastTypeTag[Spore[Int, String]]])
-    pickler.pickle(s, builder)
-    val res = builder.result()
-
-    val reader = format.createReader(res.asInstanceOf[format.PickleType])
-    val up = pickler.unpickle("scala.spores.Spore[Int, String]", reader)
-    val us = up.asInstanceOf[Spore[Int, String]]
-    val res2 = us(5)
+    val res = s.pickle
+    val up = res.value.unpickle[Spore[Int, String]]
+    val res2 = up(5)
     assert(res2 == "arg: 5, c1: 10")
+
   }
 
   @Test
   def `generated nested classes`(): Unit = {
+
     val s = spore {
       implicit val p = implicitly[Pickler[(Int, List[String])]]
       implicit val u = implicitly[Unpickler[(Int, List[String])]]
       (elem: (Int, List[String]), emit: Emitter[(Int, List[String])]) =>
         if (elem._1 == 0) emit.emit(elem)
     }
+
     assert(true)
+
   }
 
   @Test
   def testSimpleSpore(): Unit = {
+
     val s: Spore[Int, Int] = spore { (x: Int) => x + 1 }
     val p = s.pickle
     val s2 = p.unpickle[Spore[Int, Int]]
     assert(s2(10) == 11)
+
   }
 
   @Test
   def testSpore2WithEnv(): Unit = {
-    val maxSize = 20
 
-    val s = spore {
+    val maxSize = 20
+    val s: Spore2WithEnv[(String, Int), Emitter[String], Unit] {type Captured = (Int, Int)} = spore {
```

Comment:
This looks to me like a different test, since the type of the spore is no longer inferred. How can we know for sure that the type of the spore did not change? Tests where the type of a spore is inferred are different from tests where the type is provided explicitly. Therefore, you should really make sure there remains a test which looks like the previous one.